servicetest/devicenode/test_devicenode_gateway.py

```diff
@@ -0,0 +1,188 @@
+                        [
+                            {"id": "devicenode", "config": GW_CONFIG_SAME_DEV_1}
+                        ])
+
+""" Following capabilities are fed to DynamicConfiguration tests to examine the case
+    when a new device is added to configuration
+"""
+test_cap_2 = Capability("double.activation.new.device.0.cap",
+                        [
+                            {"id": "devicenode", "config": GW_CONFIG_NEW_DEV_0}
+                        ])
+
+test_cap_3 = Capability("double.activation.new.device.1.cap",
+                        [
+                            {"id": "devicenode", "config": GW_CONFIG_NEW_DEV_1}
+                        ])
+
+""" Following capability is fed to DynamicConfiguration tests to examine the case
+    when exactly same device with same mode is added
+"""
+test_cap_4 = Capability("double.activation.same.device.same.mode.cap",
+                        [
+                            {"id": "devicenode", "config": GW_CONFIG_SAME_DEV_SAME_MODE}
+                        ])
+
+""" Define the manifest so it can be used by the testframework  """
+manifest = StandardManifest(TESTOUTPUT_DIR,
+                            "devicenode-test-manifest.json",
+                            [
+                                test_cap_0,
+                                test_cap_1,
+                                test_cap_2,
+                                test_cap_3,
+                                test_cap_4
+                            ])
+
+
+def service_manifests():
+    """ The agent fixture calls this function when it creates the service manifests
+        that should be used with this test module. The agent fixture expects a list
+        of StandardManifest and/or DefaultManifest objects.
+    """
+    return [manifest]
+
+# Test suites
+
+
+@pytest.mark.usefixtures("testhelper", "agent")
+class TestDeviceNode(object):
+    """ This suite tests the behavior of DeviceNodeGateway within given set of capabilities.
+        It uses the DeviceNodeGateway to assert behavior.
+
+        The tests use the Agent D-Bus interface to enable gateway configurations,
+        for a specified list of capabilities.
+    """
+
+    @pytest.mark.parametrize(("firstcap", "secondcap"), [
+        ("double.activation.same.device.0.cap", "double.activation.same.device.1.cap"),
+        ("double.activation.new.device.0.cap", "double.activation.new.device.1.cap"),
+        ("double.activation.same.device.same.mode.cap", "double.activation.same.device.same.mode.cap")
+    ])
+    def test_dynamic_configuration(self, firstcap, secondcap):
+        """ This function tests multiple activation of DeviceNodeGateway within given set
+            of capabilities.
+        """
+        try:
+            sc = Container()
+            sc.start(DATA)
+
+            sc.set_capabilities([firstcap])
+            sc.set_capabilities([secondcap])
+        finally:
+            sc.terminate()
```

Comment:
I don't see any assert anywhere, so I'm assuming the test is considered "pass" when there is no exception? This should be explained in the comment if that is  the case.